### ISSUE TITLE
Service endpoints cannot be set using shared config file

### ISSUE THREAD

**Issue by gdavison** [aws/aws-sdk-go-v2#2413]
### Describe the bug

When service endpoints are set in shared config files, they are not applied to the `aws.Config`.

```ini
[default]
services = example

[services example]
s3 =
  endpoint_url = https://s3.example.com
```

### Expected Behavior

The service endpoints would be set

### Current Behavior

The service endpoints are not set

### Reproduction Steps

Add the following test case to `TestNewSharedConfig` in `config/shared_config_test.go`

```go
"service endpoint config": {
	ConfigFilenames: []string{testConfigFilename},
	Profile:         "service-endpoints",
	Expected: SharedConfig{
		Profile: "service-endpoints",
		Services: Services{
			ServiceValues: map[string]map[string]string{
				"s3": map[string]string{
					"endpoint_url": "https://s3.example.com",
				},
			},
		},
	},
},
```

Add the following sections to `config/testdata/shared_config`

```ini
[profile service-endpoints]
services = test

[services test]
s3 =
  endpoint_url = https://s3.example.com
```

The specific test case will fail with the following diff

```
config.SharedConfig{
                ... // 32 identical fields
                IgnoreConfiguredEndpoints: nil,
                BaseEndpoint:              "",
        - 	Services: config.Services{
        - 		ServiceValues: map[string]map[string]string{"s3": {"endpoint_url": "https://s3.example.com"}},
        - 	},
        + 	Services:             config.Services{ServiceValues: map[string]map[string]string{"s3": {}}},
                S3DisableExpressAuth:      nil,
          }
```

In addition, most of the other test cases will now fail with the following diff

```
config.SharedConfig{
                ... // 32 identical fields
                IgnoreConfiguredEndpoints: nil,
                BaseEndpoint:              "",
        -       Services:                  config.Services{},
        +       Services:                  config.Services{ServiceValues: map[string]map[string]string{"s3": {}}},
                S3DisableExpressAuth: nil,
          }
```

It looks like all profiles are referencing the `[services test]` section

### AWS Go SDK V2 Module Versions Used

Using the `main` branch of `aws-sdk-go-v2`, at commit `8c02c46d48`

### Compiler and Version used

go version go1.21.0 darwin/arm64

### Operating System and version

macOS 13.4.1

**Comment by lucix-aws** [comment]
Regression from the INI parse refactor -- will be resolved by #2416. You also caught a small miss in how we load the services section - it's only supposed to be loaded into shared config if referenced in the profile via the `services` key (that's why every other test started failing with that data) - the PR includes that fix.

**Comment by lucix-aws** [comment]
This has been merged. Please be sure that both the `config` and `internal/ini` module versions are up-to-date after tomorrow's release.
